## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 495479`
Commit: [`90b804beaac4055f731d20a32f83fd7aaef99189`](https://github.com/NixOS/nixpkgs/commit/90b804beaac4055f731d20a32f83fd7aaef99189) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/90b804beaac4055f731d20a32f83fd7aaef99189..pull/495479/head))
Merge: [`b758fe50e43ac140fa88f5f8529ab1e04259955a`](https://github.com/NixOS/nixpkgs/commit/b758fe50e43ac140fa88f5f8529ab1e04259955a)

Logs: https://github.com/sikmir/nixpkgs-review-gha/actions/runs/22541291722


---
### `x86_64-linux`
<details>
  <summary>:x: 12 packages failed to build:</summary>
  <ul>
    <li>bonsai</li>
    <li>hareThirdParty.hare-compress</li>
    <li>hareThirdParty.hare-ev</li>
    <li>hareThirdParty.hare-json</li>
    <li>hareThirdParty.hare-png</li>
    <li>hareThirdParty.hare-ssh</li>
    <li>hareThirdParty.hare-toml</li>
    <li>haredo</li>
    <li>haredo.man</li>
    <li>himitsu</li>
    <li>treecat</li>
    <li>treecat.man</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>hare</li>
    <li>hare.man</li>
    <li>hareHook</li>
    <li>harec</li>
    <li>haredoc</li>
    <li>haredoc.man</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>hareThirdParty.hare-compress</summary>
<pre>118 |		append(d.bufstart, b...);
    |	        ^

compress/flate/inflate.ha:124:9: error: Cannot ignore error here

124 |		append(d.hist, b[..n]...);
    |	        ^

compress/flate/inflate.ha:303:25: error: Cannot ignore error here

303 |				append(lens, c);
    |	                        ^

compress/flate/inflate.ha:322:25: error: Cannot ignore error here

322 |				append(lens, n);
    |	                        ^

/nix/store/qihhz65dp11y2yvcp9aka1v6kgwl4fp1-harec-0.26.0/bin/harec for compress::flate (/build/source/compress/flate) exited with status 4
make: *** [Makefile:17: check] Error 255</pre>
</details>
<details>
<summary>hareThirdParty.hare-ev</summary>
<pre>83 |		append(loop.dispatch, dispatch);
   |	                                      ^

ev/+linux/loop.ha:84:42: error: Argument type (nomem | *ev::ondispatch) is not assignable to parameter type nullable *opaque

84 |		return mkreq(&amp;do_cancel, dispatch);
   |	                                         ^

ev/+linux/loop.ha:158:19: error: Cannot ignore error here

158 |		sort::sort(loop.events[..nevent], size(rt::epoll_event), &amp;sort_events);
    |	                  ^

ev/+linux/signal.ha:39:32: error: Cannot ignore error here

39 |				static append(signals, sig);
   |	                               ^

/nix/store/qihhz65dp11y2yvcp9aka1v6kgwl4fp1-harec-0.26.0/bin/harec for ev (/build/source/ev) exited with status 4
make: *** [Makefile:16: check] Error 255</pre>
</details>
<details>
<summary>hareThirdParty.hare-json</summary>
<pre>53 |		append(bucket, (strings::dup(key), val));
   |	        ^

encoding/json/value.ha:130:23: error: Return type (str | nomem) is not assignable to function result type encoding::json::value

130 |			return strings::dup(s);
    |	                      ^

encoding/json/value.ha:132:20: error: Initializer of type (nomem | []encoding::json::value) is not assignable to binding type []encoding::json::value

132 |			let new: []value = alloc([], len(v));
    |	                   ^

encoding/json/value.ha:134:25: error: Cannot ignore error here

134 |				append(new, dup(v[i]));
    |	                        ^

/nix/store/qihhz65dp11y2yvcp9aka1v6kgwl4fp1-harec-0.26.0/bin/harec for encoding::json (/build/source/encoding/json) exited with status 4
make: *** [Makefile:17: check] Error 255</pre>
</details>
<details>
<summary>hareThirdParty.hare-ssh</summary>
<pre>@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/akfxp00hqxg679p4n8413801x2njppjp-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: SHELL=/nix/store/2hjsch59amjs3nbgh7ahcfzm2bfwl8zi-bash-5.3p9/bin/bash PREFIX=/nix/store/jqbs1zs29r2h5rhychxl38cgn0n35wil-hare-ssh-0-unstable-2023-11-16
make: Nothing to be done for &#x27;all&#x27;.
Running phase: checkPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;checkPhase&quot; }
check flags: SHELL=/nix/store/2hjsch59amjs3nbgh7ahcfzm2bfwl8zi-bash-5.3p9/bin/bash PREFIX=/nix/store/jqbs1zs29r2h5rhychxl38cgn0n35wil-hare-ssh-0-unstable-2023-11-16 VERBOSE=y check
hare test
Error: Dependency missing: crypto::bcrypt
make: *** [Makefile:15: check] Error 255</pre>
</details>
<details>
<summary>hareThirdParty.hare-toml</summary>
<pre>304 |				return root;
    |	                              ^

toml/parse/parse.ha:435:33: error: Cannot ignore error here

435 |					append(hierarchy, subparent);
    |	                                ^

toml/parse/parse.ha:460:68: error: Argument type (nomem | *toml::table) is not assignable to parameter type toml::container

460 |				let parsed = parse_key(&amp;ll, lt.pos, t, root,
    |	                                                                   ^

toml/parse/parse.ha:503:68: error: Argument type (nomem | *toml::table) is not assignable to parameter type toml::container

503 |				let parsed = parse_key(&amp;ll, lt.pos, t, root,
    |	                                                                   ^

/nix/store/qihhz65dp11y2yvcp9aka1v6kgwl4fp1-harec-0.26.0/bin/harec for toml::parse (/build/source/toml/parse) exited with status 4
make: *** [Makefile:22: check_local] Error 255</pre>
</details>
<details>
<summary>haredo</summary>
<pre>
354 |		return (proc, tmpfilepath);
    |	                    ^

src/haredo.ha:399:64: error: Argument type (nomem | []str) is not assignable to parameter type []str

399 |		case let s: str =&gt; yield strings::dupall(strings::split(s, &quot;.&quot;));
    |	                                                               ^

src/haredo.ha:425:39: error: Initializer of type (str | nomem) not assignable to struct field of type str

425 |			execdir = strings::dup(path::parent(dobuf)?),
    |	                                      ^

src/haredo.ha:441:29: error: No field by this name exists for this type

441 |		stream = 0: uintptr: *io::vtable,
    |	                            ^

/nix/store/qihhz65dp11y2yvcp9aka1v6kgwl4fp1-harec-0.26.0/bin/harec for /build/source/src exited with status 4</pre>
</details>
<details>
<summary>himitsu</summary>
<pre>HAREPATH: /nix/store/33hfgcp8jj92gzia5vi73b4hb4b39p60-hare-0.26.0/src/hare/stdlib
hare: /nix/store/7naq2nlazkwz7annrr3bg3gnq738wrfz-hare-wrapper/bin/hare
hare-native: /nix/store/7naq2nlazkwz7annrr3bg3gnq738wrfz-hare-wrapper/bin/hare-native
Running phase: unpackPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/xz9f9qirwlz8pfm0dl3qcprfc5s2h6zf-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: SHELL=/nix/store/2hjsch59amjs3nbgh7ahcfzm2bfwl8zi-bash-5.3p9/bin/bash
hare build  -o himitsud cmd/himitsud/
Error: /build/source/crypto/argon2bug/argon2.ha:10:14: syntax error: Unexpected &#x27;nomem&#x27;, was expecting &#x27;name&#x27;
make: *** [Makefile:24: himitsud] Error 255</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 12 packages failed to build:</summary>
  <ul>
    <li>bonsai</li>
    <li>hareThirdParty.hare-compress</li>
    <li>hareThirdParty.hare-ev</li>
    <li>hareThirdParty.hare-json</li>
    <li>hareThirdParty.hare-png</li>
    <li>hareThirdParty.hare-ssh</li>
    <li>hareThirdParty.hare-toml</li>
    <li>haredo</li>
    <li>haredo.man</li>
    <li>himitsu</li>
    <li>treecat</li>
    <li>treecat.man</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>hare</li>
    <li>hare.man</li>
    <li>hareHook</li>
    <li>harec</li>
    <li>haredoc</li>
    <li>haredoc.man</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>hareThirdParty.hare-compress</summary>
<pre>118 |		append(d.bufstart, b...);
    |	        ^

compress/flate/inflate.ha:124:9: error: Cannot ignore error here

124 |		append(d.hist, b[..n]...);
    |	        ^

compress/flate/inflate.ha:303:25: error: Cannot ignore error here

303 |				append(lens, c);
    |	                        ^

compress/flate/inflate.ha:322:25: error: Cannot ignore error here

322 |				append(lens, n);
    |	                        ^

/nix/store/z4sj1p1mncfsgh3hd8rhyaw0sq9jc045-harec-0.26.0/bin/harec for compress::flate (/build/source/compress/flate) exited with status 4
make: *** [Makefile:17: check] Error 255</pre>
</details>
<details>
<summary>hareThirdParty.hare-ev</summary>
<pre>83 |		append(loop.dispatch, dispatch);
   |	                                      ^

ev/+linux/loop.ha:84:42: error: Argument type (nomem | *ev::ondispatch) is not assignable to parameter type nullable *opaque

84 |		return mkreq(&amp;do_cancel, dispatch);
   |	                                         ^

ev/+linux/loop.ha:158:19: error: Cannot ignore error here

158 |		sort::sort(loop.events[..nevent], size(rt::epoll_event), &amp;sort_events);
    |	                  ^

ev/+linux/signal.ha:39:32: error: Cannot ignore error here

39 |				static append(signals, sig);
   |	                               ^

/nix/store/z4sj1p1mncfsgh3hd8rhyaw0sq9jc045-harec-0.26.0/bin/harec for ev (/build/source/ev) exited with status 4
make: *** [Makefile:16: check] Error 255</pre>
</details>
<details>
<summary>hareThirdParty.hare-json</summary>
<pre>53 |		append(bucket, (strings::dup(key), val));
   |	        ^

encoding/json/value.ha:130:23: error: Return type (str | nomem) is not assignable to function result type encoding::json::value

130 |			return strings::dup(s);
    |	                      ^

encoding/json/value.ha:132:20: error: Initializer of type (nomem | []encoding::json::value) is not assignable to binding type []encoding::json::value

132 |			let new: []value = alloc([], len(v));
    |	                   ^

encoding/json/value.ha:134:25: error: Cannot ignore error here

134 |				append(new, dup(v[i]));
    |	                        ^

/nix/store/z4sj1p1mncfsgh3hd8rhyaw0sq9jc045-harec-0.26.0/bin/harec for encoding::json (/build/source/encoding/json) exited with status 4
make: *** [Makefile:17: check] Error 255</pre>
</details>
<details>
<summary>hareThirdParty.hare-ssh</summary>
<pre>@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/akfxp00hqxg679p4n8413801x2njppjp-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: SHELL=/nix/store/kysqfpylfpl8yrmgahj1sk6yrih918gl-bash-5.3p9/bin/bash PREFIX=/nix/store/0l7pxhszxbg1m220s944z12rmm0k1cw8-hare-ssh-0-unstable-2023-11-16
make: Nothing to be done for &#x27;all&#x27;.
Running phase: checkPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;checkPhase&quot; }
check flags: SHELL=/nix/store/kysqfpylfpl8yrmgahj1sk6yrih918gl-bash-5.3p9/bin/bash PREFIX=/nix/store/0l7pxhszxbg1m220s944z12rmm0k1cw8-hare-ssh-0-unstable-2023-11-16 VERBOSE=y check
hare test
Error: Dependency missing: crypto::bcrypt
make: *** [Makefile:15: check] Error 255</pre>
</details>
<details>
<summary>hareThirdParty.hare-toml</summary>
<pre>304 |				return root;
    |	                              ^

toml/parse/parse.ha:435:33: error: Cannot ignore error here

435 |					append(hierarchy, subparent);
    |	                                ^

toml/parse/parse.ha:460:68: error: Argument type (nomem | *toml::table) is not assignable to parameter type toml::container

460 |				let parsed = parse_key(&amp;ll, lt.pos, t, root,
    |	                                                                   ^

toml/parse/parse.ha:503:68: error: Argument type (nomem | *toml::table) is not assignable to parameter type toml::container

503 |				let parsed = parse_key(&amp;ll, lt.pos, t, root,
    |	                                                                   ^

/nix/store/z4sj1p1mncfsgh3hd8rhyaw0sq9jc045-harec-0.26.0/bin/harec for toml::parse (/build/source/toml/parse) exited with status 4
make: *** [Makefile:22: check_local] Error 255</pre>
</details>
<details>
<summary>haredo</summary>
<pre>
354 |		return (proc, tmpfilepath);
    |	                    ^

src/haredo.ha:399:64: error: Argument type (nomem | []str) is not assignable to parameter type []str

399 |		case let s: str =&gt; yield strings::dupall(strings::split(s, &quot;.&quot;));
    |	                                                               ^

src/haredo.ha:425:39: error: Initializer of type (str | nomem) not assignable to struct field of type str

425 |			execdir = strings::dup(path::parent(dobuf)?),
    |	                                      ^

src/haredo.ha:441:29: error: No field by this name exists for this type

441 |		stream = 0: uintptr: *io::vtable,
    |	                            ^

/nix/store/z4sj1p1mncfsgh3hd8rhyaw0sq9jc045-harec-0.26.0/bin/harec for /build/source/src exited with status 4</pre>
</details>
<details>
<summary>himitsu</summary>
<pre>HAREPATH: /nix/store/x8hkqc2rmxqqlqcw3yfx9qh50rrw3qgv-hare-0.26.0/src/hare/stdlib
hare: /nix/store/wjx66n9c4nh4gs39sd8h0478k11pc5aq-hare-wrapper/bin/hare
hare-native: /nix/store/wjx66n9c4nh4gs39sd8h0478k11pc5aq-hare-wrapper/bin/hare-native
Running phase: unpackPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/xz9f9qirwlz8pfm0dl3qcprfc5s2h6zf-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: SHELL=/nix/store/kysqfpylfpl8yrmgahj1sk6yrih918gl-bash-5.3p9/bin/bash
hare build  -o himitsud cmd/himitsud/
Error: /build/source/crypto/argon2bug/argon2.ha:10:14: syntax error: Unexpected &#x27;nomem&#x27;, was expecting &#x27;name&#x27;
make: *** [Makefile:24: himitsud] Error 255</pre>
</details>
</details>

---
### `x86_64-darwin`
:white_check_mark: *No rebuilds*

---
### `aarch64-darwin`
:white_check_mark: *No rebuilds*

--- a/pkgs/by-name/bo/bonsai/package.nix
+++ b/pkgs/by-name/bo/bonsai/package.nix
@@ -9,13 +9,13 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "bonsai";
-  version = "1.2.1";
+  version = "1.5.1";
 
   src = fetchFromSourcehut {
     owner = "~stacyharper";
     repo = "bonsai";
     rev = "v${finalAttrs.version}";
-    hash = "sha256-WAne0628lELQanUv2lg8Y9QEikZVAT7Xtkndhs8Ozjw=";
+    hash = "sha256-cAlTppDNXXZaL5bG+h4jANfFg2OBd0WqEmDBmY5IvZ8=";
   };
 
   nativeBuildInputs = [

--- a/pkgs/by-name/ha/hare/001-tzdata.patch
+++ b/pkgs/by-name/ha/hare/001-tzdata.patch
@@ -1,45 +1,60 @@
-diff --git a/time/chrono/+freebsd.ha b/time/chrono/+freebsd.ha
-index af49080e..3dde7963 100644
---- a/time/chrono/+freebsd.ha
-+++ b/time/chrono/+freebsd.ha
-@@ -2,8 +2,8 @@
- // (c) Hare authors <https://harelang.org>
-
- def LOCALTIME_PATH: str = "/etc/localtime";
--def TZDB_PATH: str = "/usr/share/zoneinfo/";
-+def TZDB_PATH: str = "@tzdata@/share/zoneinfo/";
-
+diff --git c/time/chrono/+freebsd.ha i/time/chrono/+freebsd.ha
+index 967ae0a8..1779f6d9 100644
+--- c/time/chrono/+freebsd.ha
++++ i/time/chrono/+freebsd.ha
+@@ -3,4 +3,4 @@
+ 
  // The filepath of the system's "leap-seconds.list" file, which contains UTC/TAI
  // leap second data.
 -export def UTC_LEAPSECS_PATH: str = "/var/db/ntpd.leap-seconds.list";
 +export def UTC_LEAPSECS_PATH: str = "@tzdata@/share/zoneinfo/leap-seconds.list";
-diff --git a/time/chrono/+linux.ha b/time/chrono/+linux.ha
-index 2756fd6f..1ea22385 100644
---- a/time/chrono/+linux.ha
-+++ b/time/chrono/+linux.ha
-@@ -2,8 +2,8 @@
- // (c) Hare authors <https://harelang.org>
-
- def LOCALTIME_PATH: str = "/etc/localtime";
--def TZDB_PATH: str = "/usr/share/zoneinfo/";
-+def TZDB_PATH: str = "@tzdata@/share/zoneinfo/";
-
+diff --git c/time/chrono/+linux.ha i/time/chrono/+linux.ha
+index 3394a7b4..1779f6d9 100644
+--- c/time/chrono/+linux.ha
++++ i/time/chrono/+linux.ha
+@@ -3,4 +3,4 @@
+ 
  // The filepath of the system's "leap-seconds.list" file, which contains UTC/TAI
  // leap second data.
 -export def UTC_LEAPSECS_PATH: str = "/usr/share/zoneinfo/leap-seconds.list";
 +export def UTC_LEAPSECS_PATH: str = "@tzdata@/share/zoneinfo/leap-seconds.list";
-diff --git a/time/chrono/+openbsd.ha b/time/chrono/+openbsd.ha
-index 2756fd6f..1ea22385 100644
---- a/time/chrono/+openbsd.ha
-+++ b/time/chrono/+openbsd.ha
-@@ -2,8 +2,8 @@
- // (c) Hare authors <https://harelang.org>
-
- def LOCALTIME_PATH: str = "/etc/localtime";
--def TZDB_PATH: str = "/usr/share/zoneinfo/";
-+def TZDB_PATH: str = "@tzdata@/share/zoneinfo/";
-
+diff --git c/time/chrono/+openbsd.ha i/time/chrono/+openbsd.ha
+index 3394a7b4..1779f6d9 100644
+--- c/time/chrono/+openbsd.ha
++++ i/time/chrono/+openbsd.ha
+@@ -3,4 +3,4 @@
+ 
  // The filepath of the system's "leap-seconds.list" file, which contains UTC/TAI
  // leap second data.
 -export def UTC_LEAPSECS_PATH: str = "/usr/share/zoneinfo/leap-seconds.list";
 +export def UTC_LEAPSECS_PATH: str = "@tzdata@/share/zoneinfo/leap-seconds.list";
+diff --git c/time/date/+freebsd.ha i/time/date/+freebsd.ha
+index 3b08c385..b036002a 100644
+--- c/time/date/+freebsd.ha
++++ i/time/date/+freebsd.ha
+@@ -7,4 +7,4 @@ export def LOCALTIME_PATH: str = "/etc/localtime";
+ 
+ // The filepath of the system's TZDB (Timezone Database) directory, a filetree
+ // of TZif (Time Zone Information Format) files and other related files.
+-export def TZDB_PATH: str = "/usr/share/zoneinfo";
++export def TZDB_PATH: str = "@tzdata@/share/zoneinfo";
+diff --git c/time/date/+linux.ha i/time/date/+linux.ha
+index 3b08c385..b036002a 100644
+--- c/time/date/+linux.ha
++++ i/time/date/+linux.ha
+@@ -7,4 +7,4 @@ export def LOCALTIME_PATH: str = "/etc/localtime";
+ 
+ // The filepath of the system's TZDB (Timezone Database) directory, a filetree
+ // of TZif (Time Zone Information Format) files and other related files.
+-export def TZDB_PATH: str = "/usr/share/zoneinfo";
++export def TZDB_PATH: str = "@tzdata@/share/zoneinfo";
+diff --git c/time/date/+openbsd.ha i/time/date/+openbsd.ha
+index 3b08c385..b036002a 100644
+--- c/time/date/+openbsd.ha
++++ i/time/date/+openbsd.ha
+@@ -7,4 +7,4 @@ export def LOCALTIME_PATH: str = "/etc/localtime";
+ 
+ // The filepath of the system's TZDB (Timezone Database) directory, a filetree
+ // of TZif (Time Zone Information Format) files and other related files.
+-export def TZDB_PATH: str = "/usr/share/zoneinfo";
++export def TZDB_PATH: str = "@tzdata@/share/zoneinfo";

--- a/pkgs/by-name/ha/hare/002-dont-build-haredoc.patch
+++ b/pkgs/by-name/ha/hare/002-dont-build-haredoc.patch
@@ -1,39 +1,42 @@
-diff --git a/Makefile b/Makefile
-index 2482be1f..9d58bc81 100644
---- a/Makefile
-+++ b/Makefile
+diff --git i/Makefile w/Makefile
+index d1f50f5e..d60e93f6 100644
+--- i/Makefile
++++ w/Makefile
 @@ -5,7 +5,7 @@ all:
  include config.mk
  include makefiles/$(PLATFORM).$(ARCH).mk
  
--all: $(BINOUT)/hare $(BINOUT)/haredoc docs
-+all: $(BINOUT)/hare docs
+-all: $(BINOUT)/hare-configured $(BINOUT)/haredoc docs
++all: $(BINOUT)/hare-configured docs
  
  HARE_DEFINES = \
  	-D PLATFORM:str='"$(PLATFORM)"' \
-@@ -79,11 +79,10 @@ docs: \
- 	docs/haredoc.1 \
+@@ -82,14 +82,12 @@ docs: \
+ 	docs/hare-cache.1 \
+ 	docs/hare-deps.1 \
+ 	docs/hare-tool.1 \
+-	docs/haredoc.1 \
  	docs/hare-run.1 \
  	docs/hare-test.1 \
 -	docs/haredoc.5 \
  	docs/hare-module.5
  
--MAN1 = hare hare-build hare-cache hare-deps haredoc hare-run hare-test
+-MAN1 = hare hare-build hare-cache hare-deps haredoc hare-run hare-test hare-tool
 -MAN5 = haredoc hare-module
-+MAN1 = hare hare-build hare-cache hare-deps hare-run hare-test
++MAN1 = hare hare-build hare-cache hare-deps hare-run hare-test hare-tool
 +MAN5 = hare-module
  
  bootstrap:
  	@BINOUT=$(BINOUT) ./scripts/genbootstrap
-@@ -104,7 +103,6 @@ install-cmd:
+@@ -110,7 +108,6 @@ install-cmd: all $(BINOUT)/hare-configured
  		'$(DESTDIR)$(BINDIR)' '$(DESTDIR)$(MANDIR)/man1' \
  		'$(DESTDIR)$(BINDIR)' '$(DESTDIR)$(MANDIR)/man5'
- 	install -m755 '$(BINOUT)/hare' '$(DESTDIR)$(BINDIR)/hare'
+ 	install -m755 '$(BINOUT)/hare-configured' '$(DESTDIR)$(BINDIR)/hare'
 -	install -m755 '$(BINOUT)/haredoc' '$(DESTDIR)$(BINDIR)/haredoc'
  	for i in $(MAN1); do install -m644 docs/$$i.1 '$(DESTDIR)$(MANDIR)'/man1/$$i.1; done
  	for i in $(MAN5); do install -m644 docs/$$i.5 '$(DESTDIR)$(MANDIR)'/man5/$$i.5; done
  
-@@ -115,7 +113,6 @@ install-mods:
+@@ -121,7 +118,6 @@ install-mods:
  
  uninstall:
  	rm -- '$(DESTDIR)$(BINDIR)/hare'

--- a/pkgs/by-name/ha/hare/003-hardcode-qbe-and-harec.patch
+++ b/pkgs/by-name/ha/hare/003-hardcode-qbe-and-harec.patch
@@ -1,17 +1,8 @@
-diff --git a/cmd/hare/build.ha b/cmd/hare/build.ha
-index ce19af9e..8631b325 100644
---- a/cmd/hare/build.ha
-+++ b/cmd/hare/build.ha
-@@ -36,7 +36,7 @@ fn build(name: str, cmd: *getopt::command) (void | error) = {
- 		case let ncpu: size =>
- 			yield ncpu;
- 		},
--		version = build::get_version(os::tryenv("HAREC", "harec"))?,
-+		version = build::get_version(os::tryenv("HAREC", "@harec_bin@"))?,
- 		arch = arch,
- 		platform = build::get_platform(os::sysname())?,
- 		...
-@@ -143,8 +143,8 @@ fn build(name: str, cmd: *getopt::command) (void | error) = {
+diff --git i/cmd/hare/build.ha w/cmd/hare/build.ha
+index 44b6484f..ebecfd8b 100644
+--- i/cmd/hare/build.ha
++++ w/cmd/hare/build.ha
+@@ -144,8 +144,8 @@ fn build(name: str, cmd: *getopt::command) (void | error) = {
  	set_arch_tags(&ctx.ctx.tags, arch);
  
  	ctx.cmds = ["",

--- a/pkgs/by-name/ha/hare/package.nix
+++ b/pkgs/by-name/ha/hare/package.nix
@@ -75,7 +75,7 @@ let
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "hare";
-  version = "0.24.2";
+  version = "0.26.0";
 
   outputs = [
     "out"
@@ -85,8 +85,8 @@ stdenv.mkDerivation (finalAttrs: {
   src = fetchFromSourcehut {
     owner = "~sircmpwn";
     repo = "hare";
-    rev = finalAttrs.version;
-    hash = "sha256-61lckI0F+Ez5LR/8g6ftS0W7Q/+EU/1flTDFleBg6pc=";
+    tag = finalAttrs.version;
+    hash = "sha256-3NrhnbAR0VIyr7JkAsY8NIkW8AelPSphzIYu8QMgXsU=";
   };
 
   patches = [
@@ -170,7 +170,7 @@ stdenv.mkDerivation (finalAttrs: {
     homepage = "https://harelang.org/";
     description = "Systems programming language designed to be simple, stable, and robust";
     license = lib.licenses.gpl3Only;
-    maintainers = [ ];
+    maintainers = with lib.maintainers; [ sikmir ];
     mainProgram = "hare";
     inherit (harec.meta) platforms badPlatforms;
   };

--- a/pkgs/by-name/ha/harec/package.nix
+++ b/pkgs/by-name/ha/harec/package.nix
@@ -19,13 +19,13 @@ let
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "harec";
-  version = "0.24.2";
+  version = "0.26.0";
 
   src = fetchFromSourcehut {
     owner = "~sircmpwn";
     repo = "harec";
-    rev = finalAttrs.version;
-    hash = "sha256-YCUBdPYr/44stW9k54QoUEhNkti6ULJkVBphx7xhmKo=";
+    tag = finalAttrs.version;
+    hash = "sha256-azj37C+Uw8wqy0lf3g/kB353iufY6P7Rf20aLCRp9a8=";
   };
 
   nativeBuildInputs = [ qbe ];
@@ -74,7 +74,7 @@ stdenv.mkDerivation (finalAttrs: {
     homepage = "https://harelang.org/";
     description = "Bootstrapping Hare compiler written in C for POSIX systems";
     license = lib.licenses.gpl3Only;
-    maintainers = [ ];
+    maintainers = with lib.maintainers; [ sikmir ];
     mainProgram = "harec";
     # The upstream developers do not like proprietary operating systems; see
     # https://harelang.org/platforms/

